### ISSUE TITLE
operationId usability in code generation

### ISSUE THREAD

**Issue by derberg** [asyncapi/spec#538]
tl;dr we need a way to provide `operationId` for code generation of different applications, both server and clients, or in other words consumers and producers. So there is a need for different `operationId` names depending on the "perspective" or let's simply call it the "purpose of generated code"

Currently, AsyncAPI document should be written from the perspective of the "client", the user that interacts with described API.

This perspective causes for example a confusion around the understanding of [pub/sub semantics](https://github.com/asyncapi/spec/issues/520)

Another challenge with having only one "perspective" is around the `operationId`. 

Let's take the following part of AsyncAPI document:
```yml
/travel/status:
    subscribe:
        summary: You can listen to travel info status.
        operationId: onTravelInfo #client code perspective, generated client reacts "onTravelInfo" incomming message 
        message:
           $ref: '#/components/messages/travelInfo'
```

The client, the user subscribes to `/travel/status`. The generated code gets a function `onTravelInfo`, that reacts on incoming travel info. Sounds good, right?

Challenge is that I want to not only generate a code for the client, but also the server, the server that exposed the API described with AsyncAPI document. What shall I do with `operationId`. I either rename to to something like `subTravelInfo` which is neutral, but not super clear about the purpose and will make my generated code cryptic. I could also just rename it to `sendTravelInfo` but then have a situation that I do not follow the AsyncAPI approach to describe API from client perspective, make things even more confusing.

**Comment by fmvilas** [comment]
Same problem exists with the `summary` and `description` fields. A description that is best suited for "clients" doesn't fit well when documenting your "application". It's a mess. And it's all caused by the mess with publish/subscribe. I think once it's fixed there it will be automatically fixed here too. Especially, if we split verbs into the two categories/point of view.

**Comment by fmvilas** [comment]
I think in v3 we should require one document for the client and another one for the server. In other words, we don't allow deriving one from the other. So you can't figure out the client by just switching send and receive from the "server" document to get the client one.

It needs a PR though. @derberg I'll be happy to take it if you're not in the mood.

**Comment by derberg** [comment]
@fmvilas be my guest ☺️

**Comment by fmvilas** [comment]
Done https://github.com/asyncapi/spec/pull/875

**Comment by fmvilas** [comment]
Calm down, almighty bot, calm down.
